primaryHandler.go

```diff
@@ -88,12 +90,25 @@ func (app *App) getDeviceInfo(deviceID string) ([]db.Event, error) {
 		}
 
 		var event db.Event
-		err := json.Unmarshal(record.Data, &event)
+		data, err := app.decrypter.DecryptMessage(record.Data)
+
 		if err != nil {
-			app.measures.UnmarshalFailure.Add(1.0)
-			logging.Error(app.logger).Log(logging.MessageKey(), "Failed to unmarshal event", logging.ErrorKey(), err.Error())
+			app.measures.DecryptFailure.Add(1.0)
+			logging.Error(app.logger).Log(logging.MessageKey(), "Failed to decode event", logging.ErrorKey(), err.Error())
 			continue
 		}
+
+		err = json.Unmarshal(data, &event)
+		if err != nil {
+			app.measures.DecryptFailure.Add(1.0)
+			logging.Error(app.logger).Log(logging.MessageKey(), "Failed to unmarshal decode event", logging.ErrorKey(), err.Error())
+			err = json.Unmarshal(record.Data, &event)
```

Comment:
If an event isn't encoded the way we expect, we shouldn't try something else, I think we should just error out.